jjd.c

```diff
@@ -77,15 +77,15 @@ dial(const char *host, const char *port)
 }
 
 static int
-read_line(int fd, char *buf, size_t bufsize)
+read_line(FILE *fp, char *buf, size_t bufsize)
 {
-	char c;
+	int c;
```

Comment:
Note here that `fgetc()` returns an int instead of a char, for having room for `EOF`